BankAcc/BankAccount/BankAccount/BankAccount.cs

```diff
@@ -0,0 +1,69 @@
+﻿
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BankAccountProj
+{
+    public class BankAccount
+    {
+        private DateTime creationDate;
+        private Person person;
+        private string accountNumber;
+        private string currency;
+        private string pin;
+        private double balance;
+        public List<Transaction> transactions = new List<Transaction>();
+
+        public BankAccount(DateTime creationDate, Person person, string accountNumber, string currency, string pin, double balance)
+        {
+            this.creationDate = DateTime.Now;
+            this.person = new Person();
+            this.accountNumber = accountNumber;
+            this.currency = currency;
+            this.pin = pin;
+            this.balance = balance;
+        }
+
+        public double getAccountBalance()
+        {
+            Transaction transaction = new Transaction(DateTime.UtcNow, this, "Balance check", 0);
+            transaction.addTransaction();
+            return this.balance;
+
+        }
+
+        public void deposit(double amount)
+        {
+            Transaction transaction = new Transaction(DateTime.UtcNow, this, "Deposit", amount);
```

Comment:
E prea incalcit modul in care ai creat tranzactiile. Tranzactiile trebuie sa existe independent, fara a sti de account.

Doar account trebuie sa stie de transactions. Daca arunci instanta de account in transaction class, atunci se primeste circular dependency si nu-i ok